From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igor@armbian.com>
Date: Wed, 20 Aug 2026 12:00:00 +0200
Subject: [PATCH] net: ethercat: K3 stmmac: update xsk_tx_metadata_request() to
 the 4-arg API

The AF_XDP TX-metadata helper now takes the xsk_buff_pool and a pointer to
the metadata pointer, resolving/validating the metadata internally:

  void xsk_tx_metadata_request(const struct xsk_buff_pool *pool,
                               struct xsk_tx_metadata **pmeta,
                               const struct xsk_tx_metadata_ops *ops,
                               void *priv);

Mainline drivers/net/ethernet/stmicro/stmmac/stmmac_main.c was updated to
match, but the out-of-tree EtherCAT fork still calls the old 3-arg form, so
the kernel fails to build on 6.18:

  drivers/net/ethercat/device/K3/stmmac_main-ethercat.c:2628:17: error:
    too few arguments to function 'xsk_tx_metadata_request'
  drivers/net/ethercat/device/K3/stmmac_main-ethercat.c:2628:41: error:
    passing argument 1 ... from incompatible pointer type

Pass the pool and &meta, matching mainline stmmac_main.c. Both `pool` and
`meta` are already in scope in stmmac_xdp_xmit_zc().

Signed-off-by: Igor Pecovnik <igor@armbian.com>
---
 drivers/net/ethercat/device/K3/stmmac_main-ethercat.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethercat/device/K3/stmmac_main-ethercat.c b/drivers/net/ethercat/device/K3/stmmac_main-ethercat.c
index 93d91ec..cd3fede 100644
--- a/drivers/net/ethercat/device/K3/stmmac_main-ethercat.c
+++ b/drivers/net/ethercat/device/K3/stmmac_main-ethercat.c
@@ -2625,8 +2625,8 @@ static bool stmmac_xdp_xmit_zc(struct stmmac_priv *priv, u32 queue, u32 budget)
 		meta_req.set_ic = &set_ic;
 		meta_req.tbs = tx_q->tbs;
 		meta_req.edesc = &tx_q->dma_entx[entry];
-		xsk_tx_metadata_request(meta, &stmmac_xsk_tx_metadata_ops,
-					&meta_req);
+		xsk_tx_metadata_request(pool, &meta,
+					&stmmac_xsk_tx_metadata_ops, &meta_req);
 		if (set_ic) {
 			tx_q->tx_count_frames = 0;
 			stmmac_set_tx_ic(priv, tx_desc);
--
Armbian
